dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -208,24 +282,93 @@ else if(bs.getBundlerClass().equals(HTMLPageAsContentBundler.class.getName()))
+            audit.setIndexName( UtilMethods.isSet(newIndexName) ? newIndexName :  indexName );
             audit.setJobId(jobId);
-            audit.setJobName(dataMap.getString("QUARTZ_JOB_NAME"));
+            audit.setJobName(dataMap.getString(QUARTZ_JOB_NAME));
             audit.setLangList(UtilMethods.join(languageToIndex,","));
-            audit.setPath(paths.size()>0 ? UtilMethods.join(paths,",") : "/*");
+            audit.setPath(paths.size() > 0 ? UtilMethods.join(paths,",") : "/*");
             audit.setPathInclude(include);
-            APILocator.getSiteSearchAuditAPI().save(audit);
+            siteSearchAuditAPI.save(audit);
         }
         catch(DotDataException ex) {
-            Logger.warn(this, "can't save audit data",ex);
+            Logger.error(this, "can't save audit data",ex);
         }
         finally {
             HibernateUtil.closeSession();
         }
 
         date = DateUtil.getCurrentDate();
-        ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +APILocator.getUserAPI().getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
-        AdminLogger.log(getClass(), "Job Finished", "User: " +APILocator.getUserAPI().getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
+        ActivityLogger.logInfo(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME  );
+        AdminLogger.log(getClass(), "Job Finished", "User: " +userAPI.getSystemUser().getUserId()+ "; Date: " + date + "; Job Identifier: " + SiteSearchAPI.ES_SITE_SEARCH_NAME );
+    }
+
+     private String newIndexName(){
+        return SiteSearchAPI.ES_SITE_SEARCH_NAME + StringPool.UNDERLINE
+                + ESMappingAPIImpl.datetimeFormat.format(new Date());
+     }
+
+     private IndexMetaData getIndexMetaData(String indexAlias) throws DotDataException {
+        String indexName = null;
+        boolean defaultIndex = false;
+        long recordCount = 0;
+        if (UtilMethods.isSet(indexAlias)) {
+            indexAlias = indexAlias.split("\\s+")[0];
+            final List<String> indices = siteSearchAPI.listIndices();
+            final Map<String, String> aliasMap = esIndexAPI.getAliasToIndexMap(indices);
+            indexName = aliasMap.get(indexAlias);
+            if (UtilMethods.isSet(indexName)) {
+                if (siteSearchAPI.isDefaultIndex(indexAlias)) {
+                    Logger.info(SiteSearchJobImpl.class, String.format("Index `%s` is currently Site-Search DEFAULT.",indexAlias));
+                    defaultIndex = true;
+                }
+            } else {
+                // the alias comes with an index name that is already in use.
+                if(indices.contains(indexAlias)){
+                   indexName = indexAlias;
+                   indexAlias = null;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assigning an Object to null is a code smell.  Consider refactoring.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5093862)